### DISCUSSION TITLE
Navigating IPCC reports

### DISCUSSION THREAD

**Discussion by shiwani-w** [petermr/semanticClimate#30] [Ideas]
How can sections of IPCC reports, particularly those paragraphs which consist of extensive numeric data can be made easy to understand?
Possibilities - Make simple graphs 

Petermr: Good start

Task 1 - Framework for the new summarized IPCC report

**Comment by petermr** [comment]
# IPCC reports

There are many reports created by the UN and specifically IPCC. We study 7 in detail and may bring in others:

* WG1 [Please add description from IPCC and hyperlinks to (a) IPCC site (b) WG1 in `semanticClimate` repo.
* WG2
* WG3
*
and
* SR15
* SROCC
* SRCCL

and in 2023 these were synthesised into
* SYR

(SIW please add hyperlinks to the reports)

**Reply by petermr** [reply]
# typical structure on disk

(2023-06-16)

```
.
├── misc # other documents in IPCC system
│   ├── 2018-03-Preface-3.pdf
│   ├── AR6_FS_approve.pdf
│   ├── AR6_FS_review_process.pdf
│   ├── AR6_FS_select.pdf
│   ├── html
│   └── svg
├── sr15 # 1.5 degrees
│   ├── SR15_Chapter_2_LR.pdf
│   ├── annexes
│   ├── chapters
│   ├── figures
│   ├── glossary
│   └── spm
├── srccl # climate in changing landscape
│   ├── chapters
│   ├── fm
│   ├── glossary
│   ├── spm
│   └── ts
├── srocc # ocean and cryosphere
│   ├── annexes
│   ├── chapters
│   ├── spm
│   └── ts
├── syr # synthesis report 2023
│   ├── lr
│   └── spm
├── wg1 # physical basis
│   ├── IPCC_AR6_WGI_Chapter01_SM.pdf
│   ├── annexes
│   ├── chapters
│   ├── facts
│   ├── faqs
│   ├── spm
│   └── ts
├── wg2 # adaptation
│   ├── annexes
│   ├── chapters
│   ├── faqs
│   ├── spm
│   └── ts
└── wg3 mitigation
    ├── Chapter01
    ├── Chapter01.pdf
    ├── Chapter02
    ├── Chapter02.pdf
...
    ├── Chapter17
    ├── Chapter17.pdf
    ├── Dictionary
    ├── IPCC_AR6_WGIII_FullReport.pdf
    ├── README.md
    ├── abbreviations_snh.xml
    ├── aff_org.xml
    ├── annexes
    ├── fulltext.html
    ├── fulltext_linked.html
    ├── html
    ├── ipcc_ar6_wg3_abb_dict.xml
    ├── ipcc_ar6_wg3_man_dict.xml
    ├── logs
    ├── output.csv
    ├── spm
    ├── structure.md
    ├── svg
    └── ts

```

**Reply by petermr** [reply]
# details of a single report

example for WG1

```
.
├── IPCC_AR6_WGI_Chapter01_SM.pdf
├── annexes
│   ├── acronyms.pdf
│   ├── atlas.pdf
│   ├── faqs.pdf
│   ├── glossary.pdf
│   ├── html
│   ├── monsoons.pdf
│   └── svg
├── chapters
│   ├── Chapter10.pdf
│   ├── html
│   └── svg
├── facts
│   ├── asia.pdf
│   ├── cities.pdf
│   ├── europe.pdf
│   ├── health.pdf
│   ├── mountains.pdf
│   └── water.pdf
├── faqs
│   ├── faqs.pdf
│   ├── html
│   └── svg
├── spm
│   ├── fulltext.annotations.html
│   ├── fulltext.annotations.id.html
│   ├── fulltext.pdf
│   ├── html
│   ├── svg
│   └── total_pages_groups.html
└── ts
    ├── fulltext.pdf
    ├── html
    └── svg
```

**Reply by petermr** [reply]
# typical HTML translations

```
(base) pm286macbook-2:html pm286$ tree . -L 2
.
├── acronyms
│   ├── glossary_groups.html
│   ├── glossary_statements.html
│   ├── glossary_styles.html
│   ├── groups_groups.html
│   ├── groups_statements.html
│   ├── groups_styles.html
│   ├── normalized.html
│   ├── page_1.html
│   ├── page_10.html
│   ├── page_2.html
...
│   ├── page_9.html
│   ├── styles1.html
│   └── total_pages.html
├── atlas
│   ├── groups_groups.html
│   ├── groups_statements.html
│   ├── groups_styles.html
│   ├── page_1.html
...

│   ├── tables_96.html
│   └── total_pages.html
├── faqs
│   ├── groups_groups.html
│   ├── groups_statements.html
│   ├── groups_styles.html
│   ├── page_1.html
│   ├── page_10.html
...
│   ├── page_71.html
│   ├── page_72.html
│   ├── page_8.html
│   ├── page_9.html
│   ├── styles1.html
│   ├── tables_10.html
│   ├── tables_11.html
...
│   ├── tables_9.html
│   └── total_pages.html
├── glossary
│   ├── annotated_glossary.html
│   ├── glossary_groups.html
│   ├── glossary_statements.html
│   ├── glossary_styles.html
│   ├── groups_groups.html
│   ├── groups_statements.html
│   ├── groups_styles.html
│   ├── normalized.html
│   ├── page_1.html
│   ├── page_10.html
│   ├── page_11.html
...│   ├── page_9.html
│   ├── styles1.html
│   └── total_pages.html
└── monsoons
    ├── groups_groups.html
    ├── groups_statements.html
    ├── groups_styles.html
    ├── page_1.html
    ├── page_10.html
    ...
├── page_9.html
    ├── styles1.html
    └── total_pages.html
```
5 directories, 498 files
(base) pm286macbook-2:html pm286$

**Comment by shiwani-w** [comment]
### INSTRUCTIONS FOR INSTALLING PY4AMI IN WINDOWS

_PYTHON IS NEEDED (IF NOT INSTALLED, PLEASE INSTALL PYTHON ON YOUR SYSTEM)_
(At times, versions of python greater than 3.8 have created issues while installing py4ami. So, it is better to have python 3.8 or previous versions)

**1) Check if you have Python installed or not**

- Open command propmt
- Type `python` (If python is installed it will let you know which version it is and if not you can install python from the Microsoft app store on your windows machine)

C:\Users\shiwa>python
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.

Python installed successfully

**2) Install pip on your system -`python -m ensurepip --upgrade` (Code to install)**

**3)Install py4ami - `pip install py4ami==0.0.49`**

_(Errors that can occur - Building wheels for collected packages: py4ami, PyPDF2, future, pdfminer3, pyamiimage, tkinterhtml, tkinterweb, sknw
  Building wheel for py4ami (pyproject.toml) ... error
  error: subprocess-exited-with-error_)_

OR

(No module named 'numpy.distutils._msvccompiler' in numpy.distutils; trying from distutils
        error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/)

OR
      
       
      _(error: could not create 'build\lib\py4ami\resources\projects\oil4\PMC5344628\sections\1_body\1_2._results_and_discussion\1_2.1._chemical_composition\2_2.1.2._gc-msd_and_gc-fid_analy': The filename or extension is too long_)

---------LONG PATH NAME ERROR -----To solve the error follow the commands given on this url - https://stackoverflow.com/questions/71477633/the-filename-or-extension-is-too-long-while-installing-confluent-kafka_

STEPS TO BE FOLLOWED --------

a) Click Win+R
b) Type regedit and press Enter
c) Go to Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem
d) Edit or create a key named LongPathsEnabled (type: REG_DWORD)
e) Enter 1 as its value and press OK.
f) Restart your system and try again. It should work now.

**4) After making the  error you can run the same command ---- `pip install py4ami==0.0.49`**

INSTALLATION SUCCESFULL WITH WARNING MESSAGE ---

WARNING: The script py4ami.exe is installed in 'C:\Users\shiwa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed py4ami-0.0.49

TO SORT THE WARNING ISSUE CHANGE THE PATH BY WRITING cd IN THE NEXT COMMAND LINE AND COPY PASTE THE PATH FROM THE WARNING MESSAGE --- 
----cd C:\Users\shiwa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts----

POSSIBLE ERROR ---  Resource stopwords not found ---

SOLUTION -- python -m nltk.downloader stopwords

CODE OUTPUT ---

C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\runpy.py:127: RuntimeWarning: 'nltk.downloader' found in sys.modules after import of package 'nltk', but prior to execution of 'nltk.downloader'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
[nltk_data] Downloading package stopwords to
[nltk_data]     C:\Users\shiwa\AppData\Roaming\nltk_data...
[nltk_data]   Unzipping corpora\stopwords.zip.

**5) Check if py4ami is installed**

CODE  `C:\Users\shiwa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts>py4ami`

CODE OUTPUT --- WARNING:symbol.ini: environment variable $PYAMI_HOME not set, defaulting to bundled config.ini in C:\Users\shiwa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\py4ami
WARNING:pyami:No args, running --help
usage: py4ami [-h] [-v] {DICT,GUI,HTML,IPCC,PDF,PROJECT} ...

Py4AMI: create, manipulate, use CProject
----------------------------------------

Py4AMI is the largest collection of functionality in the AMI system.It contains executable code and libraries to manage complex documents.
A key structure is the corpus (CProject directory) which contains a list of subdirectories (CTrees) which themselves contain many different document features (text, tables, images, graphics.
Py4AMI can create, fill, manipulate, transform many of the components including PDF, HTML, TXT, images, CSV.

The subcommands:

  DICT <options>      # create/edit/search dictionaries
  GUI <options>       # run tkinter GUI (prototype)
  HTML <options>      # create/edit HTML
  IPCC <options>      # customised IPCC tools
  PDF <options>       # convert PDF into HTML and images
  PROJECT <options>   # create and transform a corpus of documents

After installation, run
  py4ami <subcommand> <options>

Examples (# foo is a comment):
  py4ami        # runs help
  py4ami -h     # runs help
  py4ami PDF -h # runs PDF help
  py4ami PDF --infile foo.pdf --outdir bar/ # converts PDF to HTML
  py4ami PROJECT --project foodir/ # converts all PDF in foodir to CTrees
  py4ami IPCC --pdf2html file/ # converts pdf file to html

----------------------------------------

positional arguments:
  {DICT,GUI,HTML,IPCC,PDF,PROJECT}
                        subcommands

optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show version 0.0.49

run:
        py4ami <subcommand> <args>
          where subcommand is in   {DICT,GUI,HTML,PDF,PROJECT} and args depend on subcommand

---- PY4AMI installed successfully---- '''

**Comment by petermr** [comment]
Great!

What time are we talking today?

On Fri, Jul 7, 2023 at 8:01 AM shiwani-w ***@***.***> wrote:

> INSTRUCTIONS FOR INSTALLING PY4AMI IN WINDOWS
>
> *PYTHON IS NEEDED (IF NOT INSTALLED, PLEASE INSTALL PYTHON ON YOUR SYSTEM)*
> (At times, versions of python greater than 3.8 have created issues while
> installing py4ami. So, it is better to have python 3.8 or previous versions)
>
> *1) Check if you have Python installed or not*
>
>    - Open command propmt
>    - Type python (If python is installed it will let you know which
>    version it is and if not you can install python from the Microsoft app
>    store on your windows machine)
>
> C:\Users\shiwa>python
> Python 3.8.10 (tags/v3.8.10:3d8993a, May 3 2021, 11:48:03) [MSC v.1928 64
> bit (AMD64)] on win32
> Type "help", "copyright", "credits" or "license" for more information.
>
> Python installed successfully
>
> *2) Install pip on your system -python -m ensurepip --upgrade (Code to
> install)*
>
> *3)Install py4ami - pip install py4ami==0.0.49*
>
>
>
> *(Errors that can occur - Building wheels for collected packages: py4ami,
> PyPDF2, future, pdfminer3, pyamiimage, tkinterhtml, tkinterweb, sknw
> Building wheel for py4ami (pyproject.toml) ... error error:
> subprocess-exited-with-error*)_
>
> OR
>
> (No module named 'numpy.distutils._msvccompiler' in numpy.distutils;
> trying from distutils
> error: Microsoft Visual C++ 14.0 or greater is required. Get it with
> "Microsoft C++ Build Tools":
> https://visualstudio.microsoft.com/visual-cpp-build-tools/)
>
> OR
>
>   _(error: could not create 'build\lib\py4ami\resources\projects\oil4\PMC5344628\sections\1_body\1_2._results_and_discussion\1_2.1._chemical_composition\2_2.1.2._gc-msd_and_gc-fid_analy': The filename or extension is too long_)
>
> ---------LONG PATH NAME ERROR -----To solve the error follow the commands
> given on this url -
> https://stackoverflow.com/questions/71477633/the-filename-or-extension-is-too-long-while-installing-confluent-kafka
> _
>
> STEPS TO BE FOLLOWED --------
>
> a) Click Win+R
> b) Type regedit and press Enter
> c) Go to
> Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem
> d) Edit or create a key named LongPathsEnabled (type: REG_DWORD)
> e) Enter 1 as its value and press OK.
> f) Restart your system and try again. It should work now.
>
> *4) After making the error you can run the same command ---- pip install
> py4ami==0.0.49*
>
> INSTALLATION SUCCESFULL WITH WARNING MESSAGE ---
>
> WARNING: The script py4ami.exe is installed in
> 'C:\Users\shiwa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts'
> which is not on PATH.
> Consider adding this directory to PATH or, if you prefer to suppress this
> warning, use --no-warn-script-location.
> Successfully installed py4ami-0.0.49
>
> TO SORT THE WARNING ISSUE CHANGE THE PATH BY WRITING cd IN THE NEXT
> COMMAND LINE AND COPY PASTE THE PATH FROM THE WARNING MESSAGE ---
> ----cd
> C:\Users\shiwa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts----
>
> POSSIBLE ERROR --- Resource stopwords not found ---
>
> SOLUTION -- python -m nltk.downloader stopwords
>
> CODE OUTPUT ---
>
> C:\Program
> Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.2800.0_x64__qbz5n2kfra8p0\lib\runpy.py:127:
> RuntimeWarning: 'nltk.downloader' found in sys.modules after import of
> package 'nltk', but prior to execution of 'nltk.downloader'; this may
> result in unpredictable behaviour
> warn(RuntimeWarning(msg))
> [nltk_data] Downloading package stopwords to
> [nltk_data] C:\Users\shiwa\AppData\Roaming\nltk_data...
> [nltk_data] Unzipping corpora\stopwords.zip.
>
> *5) Check if py4ami is installed*
>
> CODE
> C:\Users\shiwa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\Scripts>py4ami
>
> CODE OUTPUT --- WARNING:symbol.ini: environment variable $PYAMI_HOME not
> set, defaulting to bundled config.ini in
> C:\Users\shiwa\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.8_qbz5n2kfra8p0\LocalCache\local-packages\Python38\site-packages\py4ami
> WARNING:pyami:No args, running --help
> usage: py4ami [-h] [-v] {DICT,GUI,HTML,IPCC,PDF,PROJECT} ...
> Py4AMI: create, manipulate, use CProject
>
> Py4AMI is the largest collection of functionality in the AMI system.It
> contains executable code and libraries to manage complex documents.
> A key structure is the corpus (CProject directory) which contains a list
> of subdirectories (CTrees) which themselves contain many different document
> features (text, tables, images, graphics.
> Py4AMI can create, fill, manipulate, transform many of the components
> including PDF, HTML, TXT, images, CSV.
>
> The subcommands:
>
> DICT # create/edit/search dictionaries
> GUI # run tkinter GUI (prototype)
> HTML # create/edit HTML
> IPCC # customised IPCC tools
> PDF # convert PDF into HTML and images
> PROJECT # create and transform a corpus of documents
>
> After installation, run
> py4ami
>
> Examples (# foo is a comment):
> py4ami # runs help
> py4ami -h # runs help
> py4ami PDF -h # runs PDF help
> py4ami PDF --infile foo.pdf --outdir bar/ # converts PDF to HTML
> py4ami PROJECT --project foodir/ # converts all PDF in foodir to CTrees
> py4ami IPCC --pdf2html file/ # converts pdf file to html
> ------------------------------
>
> positional arguments:
> {DICT,GUI,HTML,IPCC,PDF,PROJECT}
> subcommands
>
> optional arguments:
> -h, --help show this help message and exit
> -v, --version show version 0.0.49
>
> run:
> py4ami
> where subcommand is in {DICT,GUI,HTML,PDF,PROJECT} and args depend on
> subcommand
>
> ---- PY4AMI installed successfully---- '''
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/semanticClimate/discussions/30#discussioncomment-6381200>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS3VDPRCK3FSWR5TMQDXO6X53ANCNFSM6AAAAAAZJBCNBQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Reply by shiwani-w** [reply]
1500 IST

**Comment by petermr** [comment]
great. I'll open the Zoom at about 1430 IST and the rest of the team can
check that software works and you can join at 1500 IST.
Next week Simon is planning how we present this to FSCI!

On Fri, Jul 7, 2023 at 8:18 AM shiwani-w ***@***.***> wrote:

> 1500 IST
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/semanticClimate/discussions/30#discussioncomment-6381457>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCSYLQUIZ4MNFOW6QBITXO6Z4BANCNFSM6AAAAAAZJBCNBQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Reply by shiwani-w** [reply]
Sure!
